src/components/SearchBox/onHighlightFactory.ts

```diff
@@ -15,7 +15,23 @@ const getElementType = (osmType: string) => {
   }
 };
 
-export const getSkeleton = ({ geocoder }: GeocoderOption): Feature => {
+export const getSkeleton = (
+  option: GeocoderOption | HistoryOption,
+): Feature => {
+  if (option.type === 'history') {
+    const { history } = option;
+    return {
+      center: history.geometry.coordinates,
+      type: 'Feature',
```

Comment:
Would it be possible to create another function getHistorySkeleton and place it in the `src/components/SearchBox/options/history.tsx` file?

btw, I think skeleton from search is not much used nowadays. It could be enough to use object with `type`+`center` only, as these are the only keys `useUpdatePreviewMarker` works with.